core/event/manager.go

```diff
@@ -0,0 +1,37 @@
+// Package event provides a basic API for app modules to emit events.
+package event
+
+import (
+	"context"
+
+	"google.golang.org/protobuf/runtime/protoiface"
+)
+
+// Service represents an event service which can retrieve and set an event manager in a context.
+// event.Service is a core API type that should be provided by the runtime module being used to
+// build an app via depinject.
+type Service interface {
+	// GetEventManager returns the event manager for the context or a no-op event manager if one isn't attached.
+	GetEventManager(context.Context) Manager
+}
+
+// Manager represents an event manager.
+type Manager interface {
+	// Emit emits typed events to both clients and state machine listeners
+	// (if supported). These events MUST be emitted deterministically
+	// and should be assumed to be part of blockchain consensus.
```

Comment:
```suggestion
```

I think we should wait to mention consensus till we know how and when we will do it

Reply:
removed

Reply:
would a user emit a consensus event and client event separately?

Reply:
All consensus events would also be client events. But client-only events would not be part of consensus and could safely be added in point releases.

It's basically saying, now events might be in consensus so consider them breaking, but if you need a backdoor to send something only to clients, here it is

Reply:
yea lets add it. I guess for now it would be either proto.message or kv?

Reply:
In other words, consensus events are a sub-set of client events, but not all client events are part of consensus.

Reply:
Maybe we should clarify this in the godocs? Otherwise this LGTM

Reply:
Yes we should. Here's what I propose:
- EmitProtoEvent - state that these should be emitted deterministically and might be part of consensus and that clients will also receive
- EmitKVEvent - only for clients, not part of consensus - my rationale is that I don't think there provide a great API for event listening
- EmitProtoEventNonConsensus - only emitted to clients, not part of consensus

Reply:
We agreed on this during today's SDK call.